### PR TITLE
add taplo

### LINKED ISSUE #354: [New server request] taplo-lsp
**Which server would you like to request to be added?**

- Server name: `taplo-lsp`
- Server homepage (e.g., GitHub page):
  - https://taplo.tamasfe.dev/lsp/
  - https://github.com/tamasfe/taplo

**Which languages does this server target?**

- Languages: `toml`

**How is this server distributed?**

- Is the server distributed through a standardized channel (such as release files, npm, pip, etc.)? (leave empty if unknown):
  - [`cargo install`](https://crates.io/crates/taplo-lsp)
- Is this server added in [lspconfig](https://github.com/neovim/nvim-lspconfig)?:
  - https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#taplo

### PR DESCRIPTION
**Pull request by williamboman**
Resolves #354.

### CODE REVIEW

**Comment by williamboman** [comment]
I'm struggling to install taplo-lsp on my MacBook M1, due to https://github.com/tamasfe/taplo/issues/197. It does however work in a fresh `rust` Docker container.

File: lua/nvim-lsp-installer/installers/cargo.lua [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            args = args,
+            stdio_sink = ctx.stdio_sink,
+        }, callback)
+    end
+end
+
+---@param root_dir string The directory to resolve the executable from.
+---@param executable string
+function M.executable(root_dir, executable)
+    return path.concat { root_dir, "bin", executable }
```
Comment (williamboman) [inline lua/nvim-lsp-installer/installers/cargo.lua:30]: TODO: Verify on Windows

File: lua/nvim-lsp-installer/servers/taplo/init.lua [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+return function(name, root_dir)
+    return server.Server:new {
+        name = name,
+        root_dir = root_dir,
+        languages = { "toml" },
+        homepage = "https://taplo.tamasfe.dev/lsp/",
+        installer = cargo.crates { "taplo-lsp" },
+        default_options = {
+            cmd = cargo.executable(root_dir, "taplo-lsp"),
```
Comment (zmtq05) [inline lua/nvim-lsp-installer/servers/taplo/init.lua:12]: image

I think it should be table.

Comment (zmtq05) [reply lua/nvim-lsp-installer/servers/taplo/init.lua:12]:
```suggestion
            cmd = { cargo.executable(root_dir, "taplo-lsp") },
```

Comment (zmtq05) [reply lua/nvim-lsp-installer/servers/taplo/init.lua:12]:
```suggestion
            cmd = { cargo.executable(root_dir, "taplo-lsp"), "run" },
```

https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#taplo

Comment (williamboman) [reply lua/nvim-lsp-installer/servers/taplo/init.lua]: Ah thanks

**Pull request merged**